### ISSUE TITLE
App icon with Windows appx build not working as it should according to the documentation

### ISSUE THREAD

**Issue by mathias-berg** [electron-userland/electron-builder#7811]
* **Electron-Builder Version**: 24.6.4
* 

* **Node Version**: 18.15.0

* **Electron Version**: 26.2.1
* **Electron Type (current, beta, nightly)**: current

* **Target**: 
Windows (app), MacOs (Mas)

Hi,

This issue is kind of strange, but very easy to reproduce. I have found a few people having this issue when searching, but not any solution that works.

The platforms that works for me as it should is MacOS and Windows (nsis build).
The one that I am struggling with is the `appx` build.

According to the documentation you should put your app icon in the `build` folder or the one that you assign as such in `package.json` at `buildResources`.
I use the default `build`.

My app icon is `icon.ico` for Windows and `icon.icns` for Mac (could as well be `icon.png`).

When building a `nsis` build, I get my expected icon, but when building for `appx`, I get the default Electron icon.
The only thing I could get some other icon than the default Electron one is if I also add an logo named `StoreLogo.png` in `build/appx`.

Look at the attached images. The one to the left is a `nsis` build and the one to the right is an `appx` build.
They are built right after each other, nothing is changed.
<img width="108" alt="Skärmavbild 2023-10-05 kl  19 32 32" src="https://github.com/electron-userland/electron-builder/assets/26097801/22b24b5b-ee9e-43af-b7ad-e5027056ecb7">

When hovering the app icons, I get these results. Here you can actually see the wanted icon in the `appx` build in the top left preview (coming from `icon.ico`).

nsis:
<img width="246" alt="Skärmavbild 2023-10-05 kl  19 32 45" src="https://github.com/electron-userland/electron-builder/assets/26097801/6e88bbb2-dcf5-46b1-be5c-3d216d891cf4">

appx:
<img width="276" alt="Skärmavbild 2023-10-05 kl  19 32 52" src="https://github.com/electron-userland/electron-builder/assets/26097801/ac1698bd-3fbd-4bef-8cc5-0d76b991a9a2">

One more thing.
When I added the `StoreLogo.png` image on the `appx` build, you will see a blue background on the image, that is a bit hard to see on this image.  I have tried to use the `backgroundColor` property set to `transparent` (in package.json), or any other color, but that doesn't seem to affect it all.

One last image from recent used apps.
<img width="111" alt="Skärmavbild 2023-10-05 kl  19 31 59" src="https://github.com/electron-userland/electron-builder/assets/26097801/d645eaf4-2fe7-4f05-a1e9-3948c300ab2c">

Why does `nsis` work as expected and why doesn't `appx`?

So either is the documentation needs an update or the implementation has some flaws.

Thanks in advance!

**Comment by mathias-berg** [comment]
Not stale, still a problem.

**Comment by PKZatWork** [comment]
I am seeing similar issues with the toolbar icon. When I build my app and run the exe in the win-upacked folder, the icon is larger than the icon that appears when running the appx build.

electron\dist\.icon-ico\icon.ico filesize is 256x256. 
electron\appx\icon.png is 512x512.
electron-builder version 24.6.3

electron-build.config.json:
...
  "win": {
    "target": [
      {
        "target": "appx",
        "arch": ["x64"]
      }
    ],
    "icon": "build/appx/StoreLogo.png",
    "electronLanguages": "en-US"
  },
...

**Comment by mathias-berg** [comment]
Still an issue!

**Comment by panther7** [comment]
Same issue.

**Comment by mmaietta** [comment]
Just to double check, can you confirm you have all the required assets for the various sizes the icons need to be rendered at?
https://www.electron.build/appx#appx-assets
>Default assets will be used for Logo, Square150x150Logo, Square44x44Logo and Wide310x150Logo if not provided. For assets marked Optional, these assets will not be listed in the manifest file if not provided.

**Comment by mathias-berg** [comment]
@panther7 this seems to be the solution, also found it earlier.
But I still think this should be more clear in the documentation.

**Comment by panther7** [comment]
Yeah... but my example works only on dark mode, if you need for both (light and dark), must be:

````sh
cp "icon16.png"  "Square44x44Logo.targetsize-16_altform-unplated.png"
mv "icon16.png"  "Square44x44Logo.targetsize-16_altform-lightunplated.png"
cp "icon24.png"  "Square44x44Logo.targetsize-24_altform-unplated.png"
mv "icon24.png"  "Square44x44Logo.targetsize-24_altform-lightunplated.png"
cp "icon32.png"  "Square44x44Logo.targetsize-32_altform-unplated.png"
mv "icon32.png"  "Square44x44Logo.targetsize-32_altform-lightunplated.png"
mv "icon44.png"  "Square44x44Logo.png"
cp "icon48.png"  "Square44x44Logo.targetsize-48_altform-unplated.png"
mv "icon48.png"  "Square44x44Logo.targetsize-48_altform-lightunplated.png"
mv "icon88.png"  "Square44x44Logo.scale-200.png"
```

It's fucking painfull.

**Comment by mathias-berg** [comment]
Yes, I created a script that does everything I need so I got the correct icon. But yes, painful indeed.

**Comment by mmaietta** [comment]
Yeahhh, dark mode was never considered (my guess is it may not even have existed when the implementation was first added)

Would anyone mind updating the doc file with your discoveries put into a PR? It'd help immensely as all I have are the comments on this issue.
Docs are located here if you're willing to pick this up!
https://github.com/electron-userland/electron-builder/blob/f4fc04acf3f206f01a5cf3030ff89684f136a0c0/pages/appx.md?plain=1#L10-L23
